### ISSUE TITLE
Fix Style/NegativeArrayIndex with refined substrings

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#14806]
Hello. :wave: Enjoying the new `Style/NegativeArrayIndex` cop but have run into a situation which throws an error when needing to use `String#[]` to grab the last character of a refined string.

## Steps to reproduce the problem

Ensure you have `Style/NegativeArrayIndex` enabled. Then run RuboCop against the following (this implementation has been truncated for brevity to only focus on the line that breaks RuboCop):

``` ruby
module Refinements
  module String
    refine ::String do
      def last minimum = 0
        return self[size - 1] if minimum.zero?
      end
    end
  end
end
```

⚠️ This is not reproducible when using the following despite being equivalent:

``` ruby
string = "demo"
string[string.size - 1]
```

## Expected behavior

I would expect the above to not throw an error but, instead, explain that you should use `self[-1]` instead. For example, here's the corrected code which causes the error to go away:

``` ruby
module Refinements
  module String
    refine ::String do
      def last minimum = 0
        return self[-1] if minimum.zero?
      end
    end
  end
end
```

Having this error be more intuitive would help a lot.

## Actual behavior

When using RuboCop to check the above implementation, I get the following error:

``` 
1 error occurred:
An error occurred while Style/NegativeArrayIndex cop was inspecting $HOME/Engineering/OSS/refinements/lib/refinements/string.rb:37:15.
Errors are usually caused by RuboCop bugs.
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.84.0 (using Parser 3.3.10.1, Prism 1.8.0, rubocop-ast 1.49.0, analyzing as Ruby 4.0, running on ruby 4.0.1) [arm64-darwin25.2.0]
  - rubocop-capybara 2.22.1
  - rubocop-disable_syntax 0.2.0
  - rubocop-packaging 0.6.0
  - rubocop-performance 1.26.1
  - rubocop-thread_safety 0.7.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```
